### PR TITLE
Neko soft deprecate : split off conditional compilation into separate impl

### PR DESCRIPTION
**Pull request by jdonaldson**
Here's a "soft deprecate" effort for neko code in std lib.

Neko shows up in a lot of std lib files as a conditional (`#if neko`).  However, neko isn't being actively developed at present, so the extra markup dedicated to it can be construed as clutter.

This PR identifies places in the std lib where neko is specified (uniquely) as a conditional on a base std implementation, and splits the neko-specific code into a neko-specific implementation.

This is primarily an aesthetic change, so I'm including a number of old and new people on the review to weigh in.

### CODE REVIEW

**Comment by Gama11** [comment]
This seems to look good in all cases except `JsonPrinter`, where it creates quite a bit of code duplication?

**Comment by back2dos** [comment]
Well, I'm not sure what this solves, but why not.

As for the code duplication, perhaps string quoting could be extracted to a separate class?

```haxe
package haxe.format;

class JsonHelper {
  static public function quoteString(string:String, out:Buf):Void;
}
```

Then that could be overridden by platform. Added bonus: by handing string serialization off to the native encoders, it could be used to speed up serialization on Python in general, and on JS and PHP for `-D haxeJSON`.

File: std/haxe/Timer.hx [hunk 1]
```diff
@@ -172,8 +172,6 @@ class Timer {
 	public static inline function stamp():Float {
 		#if flash
 		return flash.Lib.getTimer() / 1000;
-		#elseif (neko || php)
```
Comment (RealyUniqueName) [inline std/haxe/Timer.hx:175]: I don't see `Timer.hx` with neko implementation. Did you forget it?
Also, don't remove php conditional, pls )

Comment (RealyUniqueName) [reply std/haxe/Timer.hx:175]: Oh, it's just a duplicated code. Nvm then.

**Review by RealyUniqueName — APPROVED** [review]

File: std/haxe/Timer.hx (hunk 1 above)
Comment (jdonaldson) [reply std/haxe/Timer.hx:175]: Yep, should've called that out.  Both targets are handled by the default case.

**Comment by jdonaldson** [comment]
> This seems to look good in all cases except `JsonPrinter`, where it creates quite a bit of code duplication?

I agree, I might take @back2dos 's recommendation there and just split off a helper method.

**Review by Simn — APPROVED** [review]

**Comment by ncannasse** [comment]
I think we should leave JsonPrinter as it ;)

**Comment by jdonaldson** [comment]
Yeah I’m leaning that way now too.  I’ll leave it alone for now.

**Comment by jdonaldson** [comment]
I'll drop JsonPrinter since it causes some duplicate code issues, and resolving those complicates the intent of this PR too much.

**Comment by ncannasse** [comment]
There's a bug because you have removed code from haxe.Timer that applied to both neko and php targets, without replacements.

**Comment by kLabz** [comment]
Shouldn't it be handled by the `#elseif sys`, like it was said in the comment there?
https://github.com/HaxeFoundation/haxe/pull/9350/files/689c5ad9cc46113ad88f75b81b029eb34c62ad9f#diff-b44b9c0a3881444b2b842cf68d00ed3c

**Comment by jdonaldson** [comment]
Yeah, the default sys case should handle those two.

**Pull request merged**